File: src/cod.c
#include <stdio.h>
#define FUNCAO(x) (1.0/(x))

double func(double x){
    return 1.0/x;
}

double deltaX(int b, int a, int n){
    return (double)(b-a)/n;
}

double integralPorPontoMedio(int a, int b, int qntQuadrados, double (*funcao)(double)){
    double dX = deltaX(b,a,qntQuadrados);

    double resultado=0;
    
    for(double i=(2*a+dX)/2; i<=b;i+=dX){
        resultado+=funcao(i);
    }

    return resultado*dX;
}
void main(){
    printf("%lf\n", integralPorPontoMedio(1,2,5,func));
}

File: code-Python/cod.py
def func(x):
    return (1/x)

def delt_x(b,a,n):
    return (b-a)/n

def frange(start, stop, step):
    current = start
    while current < stop:
        yield current
        current += step

def integralPorPontoMedio(a, b, quantidadeDeQuadrados, funcao):
    dX=delt_x(b,a,quantidadeDeQuadrados)
    resultado=0
    for i in frange((2*a+dX)/2, b, dX):
        resultado+=funcao(i)
    return resultado*dX

print(integralPorPontoMedio(1,2,5,func))
